invenio_records_resources/services/records/service.py

```diff
@@ -467,11 +464,8 @@ def rebuild_index(self, identity, uow=None):
 
         Note: Skips (soft) deleted records.
         """
-        for rec_meta in self.record_cls.model_cls.query.all():
-            rec = self.record_cls(rec_meta.data, model=rec_meta)
-
-            if not rec.is_deleted:
-                self.indexer.index(rec)
+        records = self.record_cls.model_cls.query.filter_by(is_deleted=False).all()
+        self.indexer.bulk_index([rec.id for rec in records])
```

Comment:
The bulk_index method does not take into account the deleted records. Is there a reason for that if to be removed? If not you can .filter() the list to keep the one liner or just split and append only if not deleted.

Reply:
given that the `record.is_deleted` property refers to the model's `is_deleted` flag (from [invenio-records.api.RecordBase](https://github.com/inveniosoftware/invenio-records/blob/master/invenio_records/api.py#L105)), the `query.filter_by(is_deleted=False)` should do the same but on an SQL level, no?
if you prefer, we can move this filter back to the list comprehension, i just thought it would be nice to exclude those records already in the query.

Reply:
I think the point is to also reindex the deleted records, but please correct me if I am wrong @ppanero

Reply:
@max-moser, you are correct I missed the `filter_by`. Should be good. There is not deleted filter in ES so we do not want to index them.